apigee_api_catalog.module

```diff
@@ -40,3 +43,32 @@ function apigee_api_catalog_help($route_name, RouteMatchInterface $route_match)
     default:
   }
 }
+
+/**
+ * Implements hook_jsonapi_ENTITY_TYPE_filter_access() for 'apidoc'.
+ */
+function apigee_api_catalog_jsonapi_apidoc_filter_access(EntityTypeInterface $entity_type, AccountInterface $account) {
+  // Allow admins to see all entities.
+  if ($admin_permission = $entity_type->getAdminPermission()) {
+    if ($account->hasPermission($admin_permission)) {
+      return ([
+        JSONAPI_FILTER_AMONG_ALL => AccessResult::allowedIfHasPermission($account, $admin_permission),
```

Comment:
This can just return `JSONAPI_FILTER_AMONG_ALL => AccessResult::allowed()->cachePerPermissions()`,because the _if_ condition already validated the account has the required permission.